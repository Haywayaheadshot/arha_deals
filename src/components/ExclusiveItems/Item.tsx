import React, { FunctionComponent } from "react";
import { RiEyeCloseLine } from "react-icons/ri";

interface ItemProps {
  closeModal: () => void;
}

const Item: FunctionComponent<ItemProps> = ({ closeModal }) => {
  return (
    <div className="p-10 bg-primary rounded-md h-auto">
      <section className="w-full flex flex-row justify-end">
        <RiEyeCloseLine size={30} onClick={closeModal} />
      </section>
      <section>
        <h1 className="text-4xl text-center" data-testid="item-info">
          Item Name
        </h1>
        <p className="py-3">
          This item is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a. is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a. This item is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a is a is a is a is a is a is a is a is a is a
          is a is a is a is a is a.
        </p>
      </section>
    </div>
  );
};

export default Item;
