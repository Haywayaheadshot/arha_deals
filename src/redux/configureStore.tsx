import { createLogger } from "redux-logger";
import { configureStore } from "@reduxjs/toolkit";
import newsSlice from "./news/News";

const logger = createLogger();

const store = configureStore({
  reducer: {
    news: newsSlice,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
