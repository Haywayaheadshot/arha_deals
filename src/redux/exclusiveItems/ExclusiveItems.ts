import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { ExclusiveItemsState, Items } from "./types";
import defaultState from "./defaultState";

const timeout = 3000;

const initialState: ExclusiveItemsState = {
  items: defaultState,
};

const exclusiveItemsSlice = createSlice({
  name: "exclusiveItems",
  initialState,
  reducers: {
    fetchItems: (state, action: PayloadAction<Items[]>) => {
      setTimeout(() => {
        state.items = action.payload;
      }, timeout);
    },
  },
});

export const { fetchItems } = exclusiveItemsSlice.actions;

export default exclusiveItemsSlice.reducer;
