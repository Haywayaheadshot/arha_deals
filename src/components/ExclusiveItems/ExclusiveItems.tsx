import React, { useState } from "react";
import Item from "./Item";
import LoadingAnimation from "../Shared/LoadingAnimation";
import { useSelector } from "react-redux";
import { ExclusiveItemsState } from "../../redux/exclusiveItems/types";

const ExclusiveItems = () => {
  const [showModal, setShowModal] = useState(false);
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  const items = useSelector((state: ExclusiveItemsState) => state.items.items);

  const handleCloseModal = () => {
    setShowModal(false);
  };

  const handleShowDetails = () => {
    setShowModal(true);
  };

  return (
    <div className="py-5">
      <section className="pt-10">
        <h1 className="text-3xl text-center">Exclusive Items</h1>
        <p className="px-5 text-center py-3">
          Please note that this section is for collectors and is rearely
          populated. If there is an item, be rest assured the authencity is
          100%.
        </p>
      </section>
      {Array.isArray(items) ? (
        <section className="px-4 flex flex-col justify-center items-center gap-8">
          {items.map((item) => (
            <section
              className="border-tertiary border-2 rounded-md"
              key={item.id}
            >
              <div onClick={() => handleShowDetails()}>
                <ul className="flex flex-row justify-center gap-4 bg-secondary">
                  <li className="bg-yellow-300 rounded-md px-2 py-1 text-primary">
                    {item.itemName}
                  </li>
                  <li className="bg-tertiary rounded-md px-2 py-1 text-primary">
                    {item.availableSince}
                  </li>
                </ul>
                <img src={item.image_url} alt="Gem Stone" />
                <ul className="flex flex-row justify-center gap-4 bg-secondary">
                  <li className="bg-yellow-300 rounded-md px-2 py-1 text-primary">
                    {item.price}
                  </li>
                  <li className="bg-tertiary rounded-md px-2 py-1 text-primary">
                    {item.stock}
                  </li>
                </ul>
              </div>
              <a
                href={`https://api.whatsapp.com/send?phone=+233244041362&text=Hello%2C%0AI%20am%20interested%20in%20buying%20${item.itemName}.%0AIt%20has%20the%20id%3D${item.id}.%0AIt's%20price%20is%20${item.price}.%0A%0APlease%20get%20back%20to%20me%20as%20soon%20as%20you%20can.`}
                target="_blank"
                rel="noreferrer noopener"
              >
                <button
                  type="button"
                  className="text-xl text-center bg-red-200 text-tertiary font-semibold w-full"
                >
                  BUY
                </button>
              </a>
            </section>
          ))}
        </section>
      ) : (
        <LoadingAnimation />
      )}
      {showModal && (
        <div className="fixed bg-secondary top-0 left-0 right-0 flex h-full">
          <div className="p-10 h-full w-full flex justify-center overflow-y-auto snap-y">
            <Item closeModal={handleCloseModal} />
          </div>
        </div>
      )}
    </div>
  );
};

export default ExclusiveItems;
