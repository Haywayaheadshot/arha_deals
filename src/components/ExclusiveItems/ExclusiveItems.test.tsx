import React from "react";
import { Provider } from "react-redux";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import ExclusiveItems from "./ExclusiveItems";
import { Store, AnyAction } from "redux";
import { create } from "react-test-renderer";
import "@testing-library/jest-dom";

const mockStore = configureStore([]);

describe("ExclusiveItems", () => {
  let store: Store<unknown, AnyAction>;
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    store = mockStore({
      items: {
        items: [
          {
            id: "1",
            itemName: "item 1",
            availableSince: "2022-05-01",
            image_url: "https://example.com/image1.jpg",
            price: "100",
            stock: "10",
          },
          {
            id: "2",
            itemName: "item 2",
            availableSince: "2022-05-02",
            image_url: "https://example.com/image2.jpg",
            price: "200",
            stock: "20",
          },
        ],
      },
    });

    component = create(
      <Provider store={store}>
        <ExclusiveItems />
      </Provider>
    );
  });

  it("should render correctly", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should display the item name, available since date, image, price, and stock", () => {
    render(
      <Provider store={store}>
        <ExclusiveItems />
      </Provider>
    );

    expect(screen.getByText("item 1")).toBeInTheDocument();
    expect(screen.getByText("2022-05-01")).toBeInTheDocument();
    // expect(
    //   screen.getByRole("img", { name: `Image of ${items.itemName}` })
    // ).toBeInTheDocument();
    expect(screen.getByText("100")).toBeInTheDocument();
    expect(screen.getByText("10")).toBeInTheDocument();

    expect(screen.getByText("item 2")).toBeInTheDocument();
    expect(screen.getByText("2022-05-02")).toBeInTheDocument();
    // expect(screen.getAllByAltText("Image of item 1")[1]).toBeInTheDocument();
    expect(screen.getByText("200")).toBeInTheDocument();
    expect(screen.getByText("20")).toBeInTheDocument();
  });

  // it("should open the modal when an item is clicked", () => {
  //   const spy = jest.spyOn(window, "open");
  //   render(
  //     <Provider store={store}>
  //       <ExclusiveItems />
  //     </Provider>
  //   );

  //   // expect(screen.queryByTestId("item-info")).not.toBeInTheDocument();
  //   // screen.getByText("item 1").click();
  //   // expect(screen.getByText("Item Description")).toBeInTheDocument();

  //   fireEvent.click(screen.getByText("BUY"));
  //   expect(spy).toHaveBeenCalledWith(
  //     "https://api.whatsapp.com/send?phone=+233244041362&text=Hello%2C%0AI%20am%20interested%20in%20buying%20${item.itemName}.%0AIt%20has%20the%20id%3D${item.id}.%0AIt's%20price%20is%20${item.price}.%0A%0APlease%20get%20back%20to%20me%20as%20soon%20as%20you%20can."
  //   );
  //   spy.mockRestore();
  // });
});
