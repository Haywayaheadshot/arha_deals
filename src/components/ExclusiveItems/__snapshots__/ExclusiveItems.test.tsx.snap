// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExclusiveItems should render correctly 1`] = `
<div
  className="py-5"
>
  <section
    className="pt-10"
  >
    <h1
      className="text-3xl text-center"
    >
      Exclusive Items
    </h1>
    <p
      className="px-5 text-center py-3"
    >
      Please note that this section is for collectors and is rearely populated. If there is an item, be rest assured the authencity is 100%.
    </p>
  </section>
  <section
    className="px-4 flex flex-col justify-center items-center gap-8"
  >
    <section
      className="border-tertiary border-2 rounded-md"
    >
      <div
        data-testid="item-button"
        onClick={[Function]}
      >
        <ul
          className="flex flex-row justify-center gap-4 bg-secondary"
        >
          <li
            className="bg-yellow-300 rounded-md px-2 py-1 text-primary"
          >
            item 1
          </li>
          <li
            className="bg-tertiary rounded-md px-2 py-1 text-primary"
          >
            2022-05-01
          </li>
        </ul>
        <img
          alt="Image of item 1"
          src="https://example.com/image1.jpg"
        />
        <ul
          className="flex flex-row justify-center gap-4 bg-secondary"
        >
          <li
            className="bg-yellow-300 rounded-md px-2 py-1 text-primary"
          >
            100
          </li>
          <li
            className="bg-tertiary rounded-md px-2 py-1 text-primary"
          >
            10
          </li>
        </ul>
      </div>
      <a
        href="https://api.whatsapp.com/send?phone=+233244041362&text=Hello%2C%0AI%20am%20interested%20in%20buying%20item 1.%0AIt%20has%20the%20id%3D1.%0AIt's%20price%20is%20100.%0A%0APlease%20get%20back%20to%20me%20as%20soon%20as%20you%20can."
        rel="noreferrer noopener"
        target="_blank"
      >
        <button
          className="text-xl text-center bg-red-200 text-tertiary font-semibold w-full"
          type="button"
        >
          BUY
        </button>
      </a>
    </section>
    <section
      className="border-tertiary border-2 rounded-md"
    >
      <div
        data-testid="item-button"
        onClick={[Function]}
      >
        <ul
          className="flex flex-row justify-center gap-4 bg-secondary"
        >
          <li
            className="bg-yellow-300 rounded-md px-2 py-1 text-primary"
          >
            item 2
          </li>
          <li
            className="bg-tertiary rounded-md px-2 py-1 text-primary"
          >
            2022-05-02
          </li>
        </ul>
        <img
          alt="Image of item 2"
          src="https://example.com/image2.jpg"
        />
        <ul
          className="flex flex-row justify-center gap-4 bg-secondary"
        >
          <li
            className="bg-yellow-300 rounded-md px-2 py-1 text-primary"
          >
            200
          </li>
          <li
            className="bg-tertiary rounded-md px-2 py-1 text-primary"
          >
            20
          </li>
        </ul>
      </div>
      <a
        href="https://api.whatsapp.com/send?phone=+233244041362&text=Hello%2C%0AI%20am%20interested%20in%20buying%20item 2.%0AIt%20has%20the%20id%3D2.%0AIt's%20price%20is%20200.%0A%0APlease%20get%20back%20to%20me%20as%20soon%20as%20you%20can."
        rel="noreferrer noopener"
        target="_blank"
      >
        <button
          className="text-xl text-center bg-red-200 text-tertiary font-semibold w-full"
          type="button"
        >
          BUY
        </button>
      </a>
    </section>
  </section>
</div>
`;
